chore(build): update build process

douglasduteil opened the pull request:
Hi @guybedford 

It way harder than expected to update babel (#34).
I tried to run your `build.sh` script but it fails with : 

```sh
./build.sh: 2: ./build.sh: jspm: not found
./build.sh: 3: ./build.sh: jspm: not found
./build.sh: 4: ./build.sh: jspm: not found
./build.sh: 5: ./build.sh: jspm: not found
./build.sh: 6: ./build.sh: jspm: not found
cat: jspm_packages/github/facebook/regenerator@0.8.42/runtime.js: No such file or directory
```

So given what you wrote I try to make a standalone build script with less environnement descendances. This PR is the state of this experiment.

However, I end up with a strange error from `babel-types@6.8.1` :

```js
err        throw new Error('Module ' + name + ' not declared as a dependency of ' + entry.name);
        ^
  Error: Module ./definitions/init not declared as a dependency of /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js
      at Object.<anonymous> (/home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:329:25)
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:864:4
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:865:3
      at Object.exports.runInThisContext (vm.js:54:17)
      at Object.<anonymous> (/home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:6:43)
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:144:4
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:145:3
      at Object.exports.runInThisContext (vm.js:54:17)
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-template@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-helpers@6.8.0/lib/helpers.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-helpers@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/lib/transformation/file/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/lib/api/node.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/systemjs-build-modular-babel-helpers.js
  Error loading /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/systemjs-build-modular-babel-helpers.js

```

Do you have any idea where it comes from ?
Thanks

<br>
<br>
---
<br>
<br>

Full output: 
```sh                                                                                                                                                                                                                       
$ ./build.sh                                         
#!/bin/bash -xv

NODE_CWD=./node_modules/.bin
+ NODE_CWD=./node_modules/.bin

JSPM_CMD=${NODE_CWD}/jspm
+ JSPM_CMD=./node_modules/.bin/jspm

JSPM_VERSION=beta
+ JSPM_VERSION=beta
REGENERATOR_VERSION=0.8.0
+ REGENERATOR_VERSION=0.8.0

#

cd build-babel
+ cd build-babel

npm install jspm@${JSPM_VERSION} --cache-min=Infinity
+ npm install jspm@beta --cache-min=Infinity
/home/x/zzz/github/douglasduteil/plugin-babel/build-babel
└── jspm@0.17.0-beta.14  extraneous

npm WARN build-babel No description
npm WARN build-babel No repository field.
npm WARN build-babel No license field.
${JSPM_CMD} install --quick --log err
+ ./node_modules/.bin/jspm install --quick --log err

cp -r jspm_packages_override jspm_packages
+ cp -r jspm_packages_override jspm_packages

${JSPM_CMD} build systemjs-babel.js ../systemjs-babel-browser.js --format amd --skip-source-maps --minify --log ok
+ ./node_modules/.bin/jspm build systemjs-babel.js ../systemjs-babel-browser.js --format amd --skip-source-maps --minify --log ok
ok   Built into ../systemjs-babel-browser.js, minified, mangled as amd.

${JSPM_CMD} build systemjs-babel.js ../systemjs-babel-node.js --format amd --node --skip-source-maps --log ok
+ ./node_modules/.bin/jspm build systemjs-babel.js ../systemjs-babel-node.js --format amd --node --skip-source-maps --log ok
ok   Built into ../systemjs-babel-node.js, unminified as amd.

${JSPM_CMD} run systemjs-build-babel-helpers.js > ../babel-helpers.js
+ ./node_modules/.bin/jspm run systemjs-build-babel-helpers.js

err        throw new Error('Module ' + name + ' not declared as a dependency of ' + entry.name);
        ^
  Error: Module ./definitions/init not declared as a dependency of /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js
      at Object.<anonymous> (/home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:329:25)
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:864:4
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:865:3
      at Object.exports.runInThisContext (vm.js:54:17)
      at Object.<anonymous> (/home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:6:43)
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:144:4
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:145:3
      at Object.exports.runInThisContext (vm.js:54:17)
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-template@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-helpers@6.8.0/lib/helpers.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-helpers@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/lib/transformation/file/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/lib/api/node.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/systemjs-build-babel-helpers.js
  Error loading /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/systemjs-build-babel-helpers.js
${JSPM_CMD} run systemjs-build-modular-babel-helpers.js
+ ./node_modules/.bin/jspm run systemjs-build-modular-babel-helpers.js

err        throw new Error('Module ' + name + ' not declared as a dependency of ' + entry.name);
        ^
  Error: Module ./definitions/init not declared as a dependency of /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js
      at Object.<anonymous> (/home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:329:25)
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:864:4
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js:865:3
      at Object.exports.runInThisContext (vm.js:54:17)
      at Object.<anonymous> (/home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:6:43)
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:144:4
      at /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js:145:3
      at Object.exports.runInThisContext (vm.js:54:17)
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-types@6.8.1/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/lib/virtual-types.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/path/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-traverse@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-template@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-helpers@6.8.0/lib/helpers.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-helpers@6.8.0/lib/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/lib/transformation/file/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/lib/api/node.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/jspm_packages/npm/babel-core@6.8.0/index.js
  Evaluating /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/systemjs-build-modular-babel-helpers.js
  Error loading /home/x/zzz/github/douglasduteil/plugin-babel/build-babel/systemjs-build-modular-babel-helpers.js

(
  echo "export default (function(module) {" ;
  cat jspm_packages/github/facebook/regenerator@${REGENERATOR_VERSION}/runtime.js
  echo "return module.exports; })({exports:{}});"
) > ../regenerator-runtime.js
+ echo 'export default (function(module) {'
+ cat jspm_packages/github/facebook/regenerator@0.8.0/runtime.js
+ echo 'return module.exports; })({exports:{}});'
cd ..
+ cd ..
```

guybedford commented on build-babel/jspm.config.js:
```diff
@@ -37,7 +37,7 @@ SystemJS.config({
     "babel-preset-stage-3": "npm:babel-preset-stage-3@6.5.0",
     "core-js": "npm:core-js@1.2.6",
-    "regenerator": "github:facebook/regenerator@0.8.42",
+    "regenerator": "github:facebook/regenerator@0.8.0",
```
This might explain why `regenerator@0.8.42` was not found!?

douglasduteil replied:
Right

guybedford commented on build.sh:
```diff
@@ -1,12 +1,27 @@
+
+npm install jspm@${JSPM_VERSION} 
+${JSPM_CMD} install --quick --log err
+cp -r jspm_packages_override jspm_packages
```
Much nicer!

guybedford commented:
I've actually just been working on some updates to this build, and it is worth noting that the build is created with the latest jspm beta. Once I've posted out this release today, that should work here as expected I think?

douglasduteil commented:
Yes ! Way better :)

douglasduteil commented:
I rebased my PR.
Adding a regenerator version guessing script :

```sh
npm v regenerator@^0.8.42 version | tail -n 1 | awk '{print $2}' | tr -d "'"
```

guybedford commented on build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.9.0/lib/visit.js:
```diff
@@ -87,7 +87,7 @@ exports.visitor = {
         bodyBlockPath.node.body = innerBody;
```
This file can be removed from the commit.

douglasduteil replied:
Yes

guybedford commented on build-babel/jspm_packages/npm/babel-plugin-transform-regenerator@6.9.0/lib/util.js:
```diff
@@ -12,8 +12,8 @@ var t = _interopRequireWildcard(_babelTypes);
+  return t.memberExpression(t.identifier("regeneratorRuntime"), t.identifier(name), false);
 } /**
    * Copyright (c) 2014, Facebook, Inc.
    * All rights reserved.
```
And this one too.

douglasduteil replied:
Sure

guybedford commented on build-babel/node_modules/jspm:
```diff
@@ -1 +0,0 @@
-../../../jspm-cli
```
What change did you make to this symlink?

guybedford commented on build.sh:
```diff
@@ -1,12 +1,28 @@
+JSPM_CMD=${NODE_CWD}/jspm
+
+JSPM_VERSION=beta
+REGENERATOR_VERSION=$(npm v regenerator@^0.8.42 version | tail -n 1 | awk '{print $2}' | tr -d "'")
```
I don' think we use the `npm` version of regenerator?

douglasduteil replied:
You prefer manually entering the version here ?

douglasduteil commented on build.sh:
```diff
@@ -1,12 +1,28 @@
 (
   echo "export default (function(module) {" ;
-  cat jspm_packages/npm/regenerator-runtime@0.9.5/runtime.js
+  cat jspm_packages/github/facebook/regenerator@${REGENERATOR_VERSION}/runtime.js
```
Sorry isn't better to use the NPM version of regenerator ?

guybedford replied:
You're probably right actually
On Thu, 26 May 2016 at 12:27, Douglas Duteil <notifications@github.com>
wrote:

> In build.sh
> <https://github.com/systemjs/plugin-babel/pull/38#discussion_r64723647>:
>
> >  (
> >    echo "export default (function(module) {" ;
> > -  cat jspm_packages/npm/regenerator-runtime@0.9.5/runtime.js
> > +  cat jspm_packages/github/facebook/regenerator@${REGENERATOR_VERSION}/runtime.js
>
> Sorry isn't better to use the NPM version of regenerator ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/systemjs/plugin-babel/pull/38/files/956cd69a4506d50b97dc0b33800393791ba46a22#r64723647>
>

douglasduteil replied:
> You're probably right actually

You mean to use NPM or Github ?

guybedford replied:
npm. I'm just wondering if `regenerator-runtime` will always be in node_modules... If it's a dependency of babel-core then it will, but I'd need to check that.

douglasduteil commented:
:up: 

I
- make it run smoothly with the latest `regenerator-runtime`. So I removed `regenerator`
- renamed the `jspm_packages/npm/babel-plugin-transform-regenerator` to `jspm_packages_override/npm/babel-plugin-transform-regenerator`
- I enforce correct copy of those `jspm_packages_override` files
- Removed the `jspm` symlink to use `npm i jspm@beta`

douglasduteil commented:
I coded https://github.com/systemjs/plugin-babel/pull/41 on top of thie PR.
And It works perfectly :tada:

guybedford commented on build-babel/jspm_packages_override/npm/babel-plugin-transform-regenerator@6.8.0/lib/visit.js:
```diff
@@ -87,7 +87,7 @@ exports.visitor = {
       }
 
-      var outerFnExpr = getOuterFnExpr(path, state);
+      var outerFnExpr = getOuterFnExpr(path);
```
All these `, state` argument are needed here actually.

guybedford commented on .gitignore:
```diff
@@ -1,2 +1,7 @@
+# Ignore built files
+systemjs-babel-node.js
+regenerator-runtime.js
+systemjs-babel-browser.js
```
This is correct but I'm still not sure if I want to remove the build files just yet. I still like being able to download them from github...

guybedford commented:
👍

douglasduteil commented:
:up: 

I didn't commit the generated files so you can do it at will ;)

guybedford commented:
Where would these workflows be without you @douglasduteil..
